code/modules/mob/living/simple_animal/simple_animal.dm

```diff
@@ -701,4 +710,12 @@ var/global/list/animal_count = list() //Stores types, and amount of animals of t
 			if(!supernatural)
 				adjustBruteLoss(volume * 0.5)
 
+/mob/living/simple_animal/proc/delayedRegen()
+	set waitfor = 0
+	isRegenerating = 1
+	sleep(rand(minRegenTime, maxRegenTime)) //Don't want it being predictable
```

Comment:
Use spawn() instead, or it'll hang up the master controller (the thing that calls Life() of every mob)